java-checks/src/main/java/org/sonar/java/checks/VolatileVariablesIncrementDecrementCheck.java

```diff
@@ -71,26 +92,24 @@ private static IdentifierTree getVariableIdentifier(ExpressionTree expressionTre
     return null;
   }
 
-  private void checkBooleanToggling(@Nullable Tree tree, IdentifierTree identifier) {
-    Tree parent = tree != null ? tree.parent() : null;
-    if (parent != null) {
-      if (parent.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
-        checkBooleanToggling(parent, identifier);
-      } else if (parent.is(Tree.Kind.ASSIGNMENT)) {
-        ExpressionTree variable = ((AssignmentExpressionTree) parent).variable();
-        if ((variable.is(Tree.Kind.IDENTIFIER) && identifier.name().equals(((IdentifierTree) variable).name()))
-          || (variable.is(Tree.Kind.MEMBER_SELECT) && identifier.name().equals(((MemberSelectExpressionTree) variable).identifier().name()))) {
-          reportIssue(parent, "Use an \"AtomicBoolean\" for this field");
-        }
+  private void checkBooleanToggling(Tree tree, Symbol identifierSymbol) {
+    Tree parent = tree.parent();
+    if (parent.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+      checkBooleanToggling(parent, identifierSymbol);
+    } else if (parent.is(Tree.Kind.ASSIGNMENT)) {
+      IdentifierTree variableIdentifier = getVariableIdentifier(((AssignmentExpressionTree) parent).variable());
+      if (variableIdentifier != null && identifierSymbol.equals(variableIdentifier.symbol())) {
+        reportIssue(parent, "Use an \"AtomicBoolean\" for this field");
```

Comment:
we are missing the synchronized cases which are valid.
The following cases are synchronized, so there should not raise any issues.
```java
  public synchronized void toggleBoo() {
    boo1 = !boo1;
  }

  public void toggleBoo2() {
    synchronized (stuff) {
      boo1 = !boo1;
    }
  }
```